src/services/data_model_engine.js

```diff
@@ -173,12 +173,20 @@ export default class DataModelEngine {
     if (bundle === null) {
       throw new NotFoundError(`No bundle with id = ${bundleId} found`);
     }
-    return bundle;
+    let metadata = await this.entityRepository.getBundleMetadata(bundleId);
+    if (metadata === null) {
+      return bundle;
+    }
```

Comment:
One could argue that `this. getBundleMetadata(bundleId)` could be used here instead.

Reply:
Additionally: if there is no metadata for a bundle, should we allow to download it. I would think no.